dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```diff
@@ -337,6 +338,61 @@ public string GetJsonPropertyName(string dataModelPropertyName)
         return this._jsonPropertyNamesMap.Value[dataModelPropertyName];
     }
 
+    /// <summary>
+    /// Get the <see cref="VectorStoreRecordVectorProperty"/> for the given property name.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">When there is no match.</exception>
+    public VectorStoreRecordVectorProperty GetVectorProperty<TRecord>(VectorSearchOptions<TRecord>? searchOptions)
+    {
+        if (searchOptions is not null)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            string? propertyName = searchOptions.VectorPropertyName;
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            if (searchOptions.VectorProperty is Expression<Func<TRecord, object?>> expression)
+            {
+                if (expression.Body is ConstantExpression constant && constant.Value is string text && !string.IsNullOrWhiteSpace(text))
```

Comment:
I'm not sure we should support constants here in the property selection... An actual constant (string literal) in the code means that the property selected is fixed, and so there's no reason for the user to not express it via a MemberExpression - that's strongly-typed and better in every way...

There are likely going to be dynamic scenarios where the property to be used needs to be parameterized, e.g.:

```c#
string propertyName = // get this from a UI or something;
var actual = await sut.VectorizableTextSearchAsync(
    "A hotel with great views.",
    new() { VectorProperty = r => propertyName });
```

But that's a captured variable and not a ConstantExpression. This is something we can support, but I'd leave it for now to wait for user requests.

Reply:
I was hoping that `VectorStoreGenericDataModel` will be able to provide constants, but you are right, it's not true. I am going to remove it.